R/repair-names.R

```diff
@@ -277,18 +258,31 @@ check_unique_names <- function(x) {
   invisible(x)
 }
 
+make_names <- function(name) {
+  name <- gsub("^(|[.][.][.]|[.][.][1-9][0-9]*|[.][0-9].*)$", ".\\1", name)
```

Comment:
I found this doesn't pick up names that start with `_` and, therefore, they still get prefixed with `X`. Also a name like `.1` doesn't get enough dots at the front.

``` r
make_names("_a")
#> [1] "X_a"

make_names(".1")
#> [1] "..1"
```